contracts/series/SeriesController.sol

```diff
@@ -1203,4 +1219,127 @@ contract SeriesController is
+
+        allowedStrikeRanges[strikeUnderlyingToken] = TokenStrikeRange(
+            min,
+            max,
+            increment
+        );
+
+        emit StrikeRangeUpdated(strikeUnderlyingToken, min, max, increment);
+    }
+
+    /// @dev This assumes the existing AMM has valid tokens and is already has MINTER_ROLE
+    // TODO: move this out to a separate contract
+    function autoCreateSeriesAndBuy(
+        IMinterAmm _existingAmm,
+        uint256 _strikePrice,
+        uint40 _expirationDate,
+        bool _isPutOption,
+        uint256 _bTokenAmount,
+        uint256 _collateralMaximum
+    ) public nonReentrant {
+        // Save off the ammTokens
+        ISeriesController.Tokens memory ammTokens = ISeriesController.Tokens(
+            address(_existingAmm.getUnderlyingToken()),
+            address(_existingAmm.getPriceToken()),
+            address(_existingAmm.getCollateralToken())
+        );
+
+        // Get the bytes32 representation of the token triplet
+        bytes32 assetPair = keccak256(
+            abi.encode(
+                address(_existingAmm.getUnderlyingToken()),
+                address(_existingAmm.getPriceToken()),
+                address(_existingAmm.getCollateralToken())
+            )
+        );
+
+        // Validate the asset triplet was deployed to the amm address through the factory
+        require(
+            IAmmFactory(addressesProvider.getAmmFactory()).amms(assetPair) ==
+                address(_existingAmm),
+            "Invalid AMM"
+        );
+
+        // Validate strike - get the strike range info and ensure it is within params
+        TokenStrikeRange memory existingRange = allowedStrikeRanges[
+            ammTokens.underlyingToken
+        ];
+        require(_strikePrice >= existingRange.min, "Strike low");
+        require(_strikePrice <= existingRange.max, "Strike high");
+        require(
+            _strikePrice % existingRange.increment == 0,
+            "Strike increment invalid"
+        );
+
+        // Validate expiration has been set
+        require(
+            allowedExpirationsMap[_expirationDate] > 0,
+            "Invalid expiration"
+        );
+
+        // Create series
+        // add to the array so the Series data can be accessed in the future
+        allSeries.push(
+            createSeriesInternal(
+                _expirationDate,
+                _isPutOption,
+                ammTokens,
+                _strikePrice
+            )
+        );
+
+        // Temporary array to pass to event
+        address[] memory restrictedMinters = new address[](1);
+        restrictedMinters[0] = address(_existingAmm);
+
+        // Emit the event
+        emit SeriesCreated(
+            latestIndex,
+            ammTokens,
+            restrictedMinters,
+            _strikePrice,
+            _expirationDate,
+            _isPutOption
+        );
+
+        // Buy options
+        uint256 amtBought = IMinterAmm(_existingAmm).bTokenBuy(
```

Comment:
Do we need to do any check here to make sure we got the correct amount that the user was asking for or is this handled by btokenbuy?

Reply:
Yeah the bTokenBuy should error if the amount costs more than `_collateralMaximum` specified.